### DISCUSSION TITLE
Enable same MOTD for zsh?

### DISCUSSION THREAD

**Discussion by jrdodds** [microsoft/WSL#10871] [Q&A]
How is the message of the day run for the bash shell and how can I enable the same for the zsh?

When I start an Ubuntu window in Windows terminal for the first time in a day, a message of the day (motd) is displayed.

e.g.

```
Welcome to Ubuntu 20.04.6 LTS (GNU/Linux 5.15.133.1-microsoft-standard-WSL2 x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage
...
```

The message may also include a reminder to run `apt update`.

If I change my default shell to zsh, I no longer receive the motd.

With zsh I would like to mostly match the bash configuration.
Thanks

**Comment by shyun3** [comment]
Seems like the issue is that `zsh` does not source `/etc/profile` (because it sources its own `/etc/zsh/zprofile`). That's a problem because `/etc/profile` sources all the scripts in `/etc/profile.d`. One of these scripts is `update-motd.sh`, which manages the MOTD.

My solution was to put `emulate sh -c 'source /etc/profile'` in my `.zprofile`. There is a related issue reported here https://bugs.launchpad.net/ubuntu/+source/zsh/+bug/1800280 which is where I took the solution from.
